amazon/ion/core.py

```diff
@@ -374,28 +406,85 @@ class Timestamp(datetime):
 
     Notes:
         The ``precision`` field is passed as a keyword argument of the same name.
+        The behavior of the constructor is as follows:
+            - If microsecond/fractional_precision is specified but fractional_seconds is not, a new Decimal is
+            created from the value of microsecond and truncated to fractional_precision.
+
+            - If fractional_seconds is specified but microsecond is not, microsecond is set
+            to the value of fractional_seconds scaled up to microseconds magnitude.
+
+            - A ValueError is raised when both microsecond and fractional_seconds are specified, but their values are
+            not equivalent.
+
+            - A ValueError is raised when fractional_seconds is specified, but fractional_precision is not.
     """
-    __slots__ = [TIMESTAMP_PRECISION_FIELD, TIMESTAMP_FRACTION_PRECISION_FIELD]
+    __slots__ = [TIMESTAMP_PRECISION_FIELD, TIMESTAMP_FRACTION_PRECISION_FIELD, TIMESTAMP_FRACTIONAL_SECONDS_FIELD]
 
     def __new__(cls, *args, **kwargs):
         precision = None
         fractional_precision = None
+        fractional_seconds = None
+        MICROSECOND_ARGUMENT_INDEX = 6
+        DATETIME_MICROSECONDS = None
```

Comment:
Style nit: these should be lower snake case not upper case.

Reply:
Done.